platform/admin/configurator/admin-configurator-api/src/main/java/org/codice/ddf/admin/configurator/Configurator.java

```diff
@@ -0,0 +1,156 @@
+ * {@code update}, methods in the intended order of operation, then invoke one of the {@code commit}
+ * methods to complete the transaction.
+ * <p>
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface Configurator {
+    /**
+     * Sequentially invokes all the transaction's operations, committing their changes. If a failure
+     * occurs during the processing, a rollback is attempted of those handlers that had already been
+     * committed.
+     * <p>
+     * In the case of a successful commit, changes should be logged.
+     *
+     * @param auditMessage In the case of a successful commit, the message to pass to be audited
+     * @param auditParams  In the case of a successful commit, optional parameters to pass to be
+     *                     interpolated into the message
+     * @return report of the commit status, whether successful, successfully rolled back, or partially
+     * rolled back with errors
+     */
+    OperationReport commit(String auditMessage, String... auditParams);
+
+    /**
+     * Sequentially invokes all the transaction's operations, committing their changes. If a failure
+     * occurs during the processing, a rollback is attempted of those handlers that had already been
+     * committed.
+     *
+     * @return report of the commit status, whether successful, successfully rolled back, or partially
+     * rolled back with errors
+     */
+    OperationReport commit();
+
+    /**
+     * Starts the bundle with the given name.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String startBundle(String bundleSymName);
+
+    /**
+     * Stops the bundle with the given name.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String stopBundle(String bundleSymName);
+
+    /**
+     * Installs and starts the feature with the given name.
+     *
+     * @param featureName the name of the feature
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String startFeature(String featureName);
+
+    /**
+     * Stops the feature with the given name.
+     *
+     * @param featureName the name of the feature
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String stopFeature(String featureName);
+
+    /**
+     * Creates a property file in the system with the given set of new key:value pairs.
+     *
+     * @param propFile   the property file to create
+     * @param properties the set of key:value pairs to save to the property file
```

Comment:
No. It'll end up throwing an NPE on save. I'll update the javadoc to indicate that.